astroid/brain/brain_dataclasses.py

```diff
@@ -404,6 +407,15 @@ def _is_class_var(node: NodeNG) -> bool:
     )
 
 
+def _is_keyword_only_sentinel(node: NodeNG) -> bool:
+    """Return True if node is the KW_ONLY sentinel."""
+    return any(
+        isinstance(inferred, bases.Instance)
+        and inferred.qname() == "dataclasses._KW_ONLY_TYPE"
+        for inferred in node.infer()
```

Comment:
Do we not need to catch any exceptions?

Reply:
Yeah, we should probably use `safe_infer` here. Good call.

Although, this depends on what we want to do with failing inference on < 3.10. Perhaps we want to catch it and still return `True`?

Reply:
Does this condition need to check `if PY310_PLUS`?

Reply:
Yeah I don't really know. If we make this interpreter dependent then pylint will produce different output on different interpreters. Is that how we usually handle this?
Normally this would crash anyway as there is no KW_ONLY on lower versions in `dataclasses`.

Reply:
Yes, that's how we normally handle python feature differences.

Reply:
Okay, then this becomes pretty easy to fix. I'll update the PR :)